components/core/CMakeLists.txt

```diff
@@ -1,5 +1,6 @@
 cmake_minimum_required(VERSION 3.5.1)
 project(CLP LANGUAGES CXX C)
+set(CMAKE_CXX_STANDARD 17)
```

Comment:
Can we move this to be a `target_compile_features` for the target that required this change?
I could be wrong, but adding a `target_compile_features` with c++17 to `components/core/src/grammar/kibana/CMakeLists.txt` may solve the need for this here...